### PR TITLE
 Enable modernize-* checks and apply auto-fixes

### PR DESCRIPTION
**Pull request by Neverlord**
Note: changes have been automatically generated from `clang-tidy`.

Relates #249.

### CODE REVIEW

**Comment by Neverlord** [comment]
Btw, the last category I'd like to integrate are some of the `readability-*` checks. So there's only one more `clang-tidy` PR in the pipeline. 🙂

File: include/broker/alm/multipath.hh [hunk 1]
```diff
@@ -96,33 +96,33 @@ public:
 
   ~multipath_group();
 
-  size_t size() const noexcept {
+  [[nodiscard]] size_t size() const noexcept {
```
[the remark this answers is not in the document]
Comment (timwoj) [reply include/broker/alm/multipath.hh:99]: `[[nodiscard]]` is one of those flags that's useful in certain situations but I don't think adding it to every function signature is a good plan. Can we just disable that checker?

[the remark this answers is not in the document]
Comment (Neverlord) [reply include/broker/alm/multipath.hh]: Since you're both against it, I've disabled that check.

**Review by timwoj — APPROVED** [review]

**Comment by timwoj** [comment]
I re-merged this as 09a89303a56a76c5052788c99a3f25110a93ef89 since I needed to go back in time a little bit to fix a VERSION file issue, and git was being a pain with the merge commit in the way.

**Pull request merged**